File: ServerNodeJS/Server/constants/modelFields.js
/**
 * Centralized field names for models to eliminate magic strings
 * Ensures consistency between client and server field definitions
 * Mirrors client-side ModelFields.java
 */

// Common field names
const ID = '_id';
const ID_ALT = 'id';
const CREATED_AT = 'createdAt';
const UPDATED_AT = 'updatedAt';
const TIMESTAMP = 'timestamp';
const IS_ACTIVE = 'isActive';

// User fields
const USERNAME = 'username';
const EMAIL = 'email';
const PASSWORD = 'password';
const AVATAR = 'avatar';
const BIO = 'bio';
const FIRST_NAME = 'firstName';
const LAST_NAME = 'lastName';
const PHONE_NUMBER = 'phoneNumber';
const LAST_SEEN = 'lastSeen';
const ROLE = 'role';
const PROFILE = 'profile';
const FRIENDS = 'friends';
const BLOCKED_USERS = 'blockedUsers';

// Message fields
const CONTENT = 'content';
const TYPE = 'type';
const SENDER = 'sender';
const CHAT = 'chat';
const REPLY_TO = 'replyTo';
const ATTACHMENTS = 'attachments';
const REACTIONS = 'reactions';
const READ_BY = 'readBy';
const IS_READ = 'isRead';
const IS_DELETED = 'isDeleted';
const IS_EDITED = 'isEdited';
const EDITED_AT = 'editedAt';
const EDIT_HISTORY = 'editHistory';
const REACTION_SUMMARY = 'reactionSummary';
const CLIENT_NONCE = 'clientNonce';

// Chat fields
const NAME = 'name';
const DESCRIPTION = 'description';
const PARTICIPANTS = 'participants';
const PARTICIPANT_IDS = 'participantIds';
const LAST_MESSAGE = 'lastMessage';
const LAST_MESSAGE_TIME = 'lastMessageTime';
const UNREAD_COUNT = 'unreadCount';
const CREATED_BY = 'createdBy';
const GROUP_ID = 'groupId';
const VISIBILITY = 'visibility';
const IS_PUBLIC = 'isPublic';
const SETTINGS = 'settings';
const JOIN_REQUEST_STATUS = 'joinRequestStatus';

// Message types
const TYPE_TEXT = 'text';
const TYPE_IMAGE = 'image';
const TYPE_FILE = 'file';
const TYPE_VIDEO = 'video';
const TYPE_AUDIO = 'audio';
const TYPE_VOICE = 'voice';
const TYPE_SYSTEM = 'system';

// Chat types
const TYPE_PRIVATE = 'private';
const TYPE_GROUP = 'group';

// User roles
const ROLE_USER = 'user';
const ROLE_ADMIN = 'admin';
const ROLE_MODERATOR = 'moderator';
const ROLE_MEMBER = 'member';

// Friendship statuses
const FRIENDSHIP_NOT_FRIENDS = 'not_friends';
const FRIENDSHIP_FRIENDS = 'friends';
const FRIENDSHIP_PENDING = 'pending';
const FRIENDSHIP_SENT = 'sent';
const FRIENDSHIP_RECEIVED = 'received';
const FRIENDSHIP_NONE = 'none';

// Privacy/visibility
const VISIBILITY_PUBLIC = 'public';
const VISIBILITY_PRIVATE = 'private';

// Join request statuses
const JOIN_NONE = 'none';
const JOIN_PENDING = 'pending';
const JOIN_APPROVED = 'approved';
const JOIN_REJECTED = 'rejected';

module.exports = {
    // Common field names
    ID,
    ID_ALT,
    CREATED_AT,
    UPDATED_AT,
    TIMESTAMP,
    IS_ACTIVE,
    
    // User fields
    USERNAME,
    EMAIL,
    PASSWORD,
    AVATAR,
    BIO,
    FIRST_NAME,
    LAST_NAME,
    PHONE_NUMBER,
    LAST_SEEN,
    ROLE,
    PROFILE,
    FRIENDS,
    BLOCKED_USERS,
    
    // Message fields
    CONTENT,
    TYPE,
    SENDER,
    CHAT,
    REPLY_TO,
    ATTACHMENTS,
    REACTIONS,
    READ_BY,
    IS_READ,
    IS_DELETED,
    IS_EDITED,
    EDITED_AT,
    EDIT_HISTORY,
    REACTION_SUMMARY,
    CLIENT_NONCE,
    
    // Chat fields
    NAME,
    DESCRIPTION,
    PARTICIPANTS,
    PARTICIPANT_IDS,
    LAST_MESSAGE,
    LAST_MESSAGE_TIME,
    UNREAD_COUNT,
    CREATED_BY,
    GROUP_ID,
    VISIBILITY,
    IS_PUBLIC,
    SETTINGS,
    JOIN_REQUEST_STATUS,
    
    // Message types
    TYPE_TEXT,
    TYPE_IMAGE,
    TYPE_FILE,
    TYPE_VIDEO,
    TYPE_AUDIO,
    TYPE_VOICE,
    TYPE_SYSTEM,
    
    // Chat types
    TYPE_PRIVATE,
    TYPE_GROUP,
    
    // User roles
    ROLE_USER,
    ROLE_ADMIN,
    ROLE_MODERATOR,
    ROLE_MEMBER,
    
    // Friendship statuses
    FRIENDSHIP_NOT_FRIENDS,
    FRIENDSHIP_FRIENDS,
    FRIENDSHIP_PENDING,
    FRIENDSHIP_SENT,
    FRIENDSHIP_RECEIVED,
    FRIENDSHIP_NONE,
    
    // Privacy/visibility
    VISIBILITY_PUBLIC,
    VISIBILITY_PRIVATE,
    
    // Join request statuses
    JOIN_NONE,
    JOIN_PENDING,
    JOIN_APPROVED,
    JOIN_REJECTED
};

File: Client/app/src/main/java/com/example/chatappjava/constants/ModelFields.java
package com.example.chatappjava.constants;

/**
 * Centralized field names for models to eliminate magic strings
 * Ensures consistency between client and server field definitions
 */
public class ModelFields {
    
    // Common field names
    public static final String ID = "_id";
    public static final String ID_ALT = "id";
    public static final String CREATED_AT = "createdAt";
    public static final String UPDATED_AT = "updatedAt";
    public static final String TIMESTAMP = "timestamp";
    public static final String IS_ACTIVE = "isActive";
    
    // User fields
    public static final String USERNAME = "username";
    public static final String EMAIL = "email";
    public static final String PASSWORD = "password";
    public static final String AVATAR = "avatar";
    public static final String BIO = "bio";
    public static final String FIRST_NAME = "firstName";
    public static final String LAST_NAME = "lastName";
    public static final String PHONE_NUMBER = "phoneNumber";
    public static final String LAST_SEEN = "lastSeen";
    public static final String ROLE = "role";
    public static final String PROFILE = "profile";
    public static final String FRIENDS = "friends";
    public static final String BLOCKED_USERS = "blockedUsers";
    
    // Message fields
    public static final String CONTENT = "content";
    public static final String TYPE = "type";
    public static final String SENDER = "sender";
    public static final String CHAT = "chat";
    public static final String REPLY_TO = "replyTo";
    public static final String ATTACHMENTS = "attachments";
    public static final String REACTIONS = "reactions";
    public static final String READ_BY = "readBy";
    public static final String IS_READ = "isRead";
    public static final String IS_DELETED = "isDeleted";
    public static final String IS_EDITED = "isEdited";
    public static final String EDITED_AT = "editedAt";
    public static final String EDIT_HISTORY = "editHistory";
    public static final String REACTION_SUMMARY = "reactionSummary";
    public static final String CLIENT_NONCE = "clientNonce";
    
    // Chat fields
    public static final String NAME = "name";
    public static final String DESCRIPTION = "description";
    public static final String PARTICIPANTS = "participants";
    public static final String PARTICIPANT_IDS = "participantIds";
    public static final String LAST_MESSAGE = "lastMessage";
    public static final String LAST_MESSAGE_TIME = "lastMessageTime";
    public static final String UNREAD_COUNT = "unreadCount";
    public static final String CREATED_BY = "createdBy";
    public static final String GROUP_ID = "groupId";
    public static final String VISIBILITY = "visibility";
    public static final String IS_PUBLIC = "isPublic";
    public static final String SETTINGS = "settings";
    public static final String JOIN_REQUEST_STATUS = "joinRequestStatus";
    
    // Message types
    public static final String TYPE_TEXT = "text";
    public static final String TYPE_IMAGE = "image";
    public static final String TYPE_FILE = "file";
    public static final String TYPE_VIDEO = "video";
    public static final String TYPE_AUDIO = "audio";
    public static final String TYPE_VOICE = "voice";
    public static final String TYPE_SYSTEM = "system";
    
    // Chat types
    public static final String TYPE_PRIVATE = "private";
    public static final String TYPE_GROUP = "group";
    
    // User roles
    public static final String ROLE_USER = "user";
    public static final String ROLE_ADMIN = "admin";
    public static final String ROLE_MODERATOR = "moderator";
    public static final String ROLE_MEMBER = "member";
    
    // Friendship statuses
    public static final String FRIENDSHIP_NOT_FRIENDS = "not_friends";
    public static final String FRIENDSHIP_FRIENDS = "friends";
    public static final String FRIENDSHIP_PENDING = "pending";
    public static final String FRIENDSHIP_SENT = "sent";
    public static final String FRIENDSHIP_RECEIVED = "received";
    public static final String FRIENDSHIP_NONE = "none";
    
    // Privacy/visibility
    public static final String VISIBILITY_PUBLIC = "public";
    public static final String VISIBILITY_PRIVATE = "private";
    
    // Join request statuses
    public static final String JOIN_NONE = "none";
    public static final String JOIN_PENDING = "pending";
    public static final String JOIN_APPROVED = "approved";
    public static final String JOIN_REJECTED = "rejected";
}
